### PR TITLE
feat(lambda): detect Lambda cold starts & timeouts

### PR DESCRIPTION
**Pull request by basti1302**

### CODE REVIEW

File: packages/aws-lambda/src/wrapper.js [hunk 1]
```diff
@@ -93,8 +95,13 @@ function shimmedHandler(originalHandler, originalThis, originalArgs, _config) {
         alias,
         runtime: 'nodejs',
         functionName: context.functionName,
-        functionVersion: context.functionVersion
+        functionVersion: context.functionVersion,
+        reqId: context.awsRequestId
```
Comment (basti1302) [inline packages/aws-lambda/src/wrapper.js:99]: This is unrelated to this PR but I figured that having the request ID in the span would be actually quite helpful for correlating a trace in Instana with cloudwatch logs, so I added it.

File: packages/aws-lambda/test/integration_test/test.js [hunk 2]
```diff
@@ -1010,74 +1024,84 @@ function registerTests(handlerDefinitionPath) {
...
-        expect(span.ec).to.equal(1);
-      } else if (error === false) {
-        expect(span.data.lambda.error).to.not.exist;
-        expect(span.ec).to.equal(0);
-      } else {
-        throw new Error(`Unknown error expectation type, don't know how to verify: ${error}`);
-      }
-      verifyHeaders(span);
-    });
+    const checks = [
```
Comment (basti1302) [inline packages/aws-lambda/test/integration_test/test.js:1027]: Most of this diff comes from migrating from the `expectExactlyOneMatching(spans, callback)` syntax to `expectExactlyOneMatching(spans, [expectations])` syntax.

File: packages/aws-lambda/test/integration_test/test.js [hunk 3]
```diff
@@ -1010,74 +1024,84 @@ function registerTests(handlerDefinitionPath) {
...
+    } else if (error === false) {
+      checks.push(span => expect(span.data.lambda.error).to.not.exist);
+      checks.push(span => expect(span.ec).to.equal(0));
+    } else {
+      throw new Error(`Unknown error expectation type, don't know how to verify: ${error}`);
+    }
+    if (expectations.expectColdStart === true) {
+      checks.push(span => expect(span.data.lambda.coldStart).to.be.true);
+    } else if (expectations.expectColdStart === false) {
+      checks.push(span => expect(span.data.lambda.coldStart).to.not.exist);
```
Comment (basti1302) [inline packages/aws-lambda/test/integration_test/test.js:1077]: I believe this is actually the only new check that has been added here.

File: packages/aws-lambda/test/integration_test/test.js [hunk 4]
```diff
@@ -1010,74 +1024,84 @@ function registerTests(handlerDefinitionPath) {
...
+      span => expect(span.async).to.not.exist,
+      span => expect(span.data.http).to.be.an('object'),
+      span => expect(span.data.http.method).to.equal('GET'),
+      span => expect(span.data.http.url).to.equal(downstreamDummyUrl),
+      span =>
+        expect(span.data.http.header).to.deep.equal({
+          'x-downstream-header': 'yes'
+        }),
+      verifyHeaders
+    ]);
```
Comment (basti1302) [inline packages/aws-lambda/test/integration_test/test.js:1104]: Again, no actual change, just converting this to the preferred syntax.

**Review by willianpc — APPROVED** [review]
LGTM. I added some minor considerations you may want to check, but no show stopper.
Cheers!

**Pull request merged**